examples/ecs/dynamic.rs

```diff
@@ -0,0 +1,246 @@
+//! This example show how you can create components dynamically, spawn entities with those components
```

Comment:
All the unsafe's in this example should have safety comments.

Reply:
Yep, there are several places that need better safety comments. Definitely needs another pass, will try to get to it this weekend or some time next week.